service/topology_coordinator.cc

```diff
@@ -2187,18 +2187,13 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
         if (_topo_sm._topology.requests.empty()) {
             co_return;
         }
-        auto ts = guard.write_timestamp();
         for (auto& [id, req] : _topo_sm._topology.requests) {
-            topology_mutation_builder builder(ts);
-            topology_request_tracking_mutation_builder rtbuilder(_topo_sm._topology.find(id)->second.request_id);
-            auto node_builder = builder.with_node(id).del("topology_request");
             auto done_msg = fmt::format("Canceled. Dead nodes: {}", dead_nodes);
-            rtbuilder.done(done_msg);
+            _topo_sm.generate_cancel_request_update(muts, _feature_service, guard, id, done_msg);
```

Comment:
`_feature_service` is unused by `generate_cancel_request_update`.

It looks like we copy `done_msg` for no reason.